data/bots/lib/libbot/planning-area-shapefiles.ts

```diff
@@ -4,28 +4,37 @@ import { ShapefileUploader } from "./shapefile-uploader.ts";
 import { logDebug, logError, logInfo } from "./logger.ts";
 import { tookMs } from "./util/perf.ts";
 
-interface FileUpload {
-  url: string;
-  formField: string;
-  data: Blob;
-  fileName: string;
-  headers: [string, string][];
+enum PlanningAreaShapefileKind {
+  planningArea = 'planningArea',
+  planningGrid = 'planningGrid',
 }
 
-export class PlanningAreaShapefiles extends ShapefileUploader {
-  private url;
+interface PlanningAreaShapefile {
+  kind: PlanningAreaShapefileKind,
+  localFilePath: string,
+}
 
-  constructor(httpClient: BotHttpClient, urlPrefix: string) {
+export class PlanningAreas extends ShapefileUploader {
+  constructor(httpClient: BotHttpClient) {
     super(httpClient);
-    this.url = urlPrefix + "/api/v1/projects/planning-area/shapefile";
   }
 
-  async uploadFromFile(localFilePath: string): Promise<string> {
+  private getUrl(kind: PlanningAreaShapefileKind): string {
+    if(kind === PlanningAreaShapefileKind.planningArea) {
+      return `${this.baseUrl}/api/v1/projects/planning-area/shapefile`;
+    }
+    if(kind === PlanningAreaShapefileKind.planningGrid) {
+      return `${this.baseUrl}/api/v1/projects/planning-area/shapefile-grid`;
+    }
+    throw new Error(`Unknown planning area shapefile kind: ${kind}`);
+  }
+
+  async uploadFromFile(shapefile: PlanningAreaShapefile): Promise<string> {
```

Comment:
Could be private, as we have kind-specific ones already.

Reply:
093958a15d63c3e511cc987d6e3050925d544b1b